### PR TITLE
Show md source

### PR DESCRIPTION
**Pull request by darrenleeweber**
Fix for https://github.com/sul-dlss/argo/issues/215

### CODE REVIEW

**Comment by mejackreed** [comment]
Looks great! It would be great if we could also test this, so we know what md fields should be on the show page. Mind adding a test for this? I'm thinking a feature spec probably fits the bill.

**Comment by darrenleeweber** [comment]
:+1: for tests.  This fix is the most direct solution, given the current data manipulations.  This code is slightly opaque - it depends on how the foxml gets indexed and then how the Solr index is translated to a display string.  It's not clear to me yet whether there are already tests on the index mapping (if not, need to start there).  I'm not sure whether to test the controller or the view or both (and whether blacklight has tooling to ease the testing of Solr -> view mappings).

**Comment by mejackreed** [comment]
Yep, the MD solr fields are scatter shot through out the code. My first thought is that a feature spec captures that. Most likely the indexing code is not in Argo but in Dor Services. Something along the lines of:

```
visit catalog_path 'druid:somedruid'
within 'dl.document-metadata' do
  expect(page).to have_css 'dd', text: 'stuff'
  expect(page).to have_css 'dt', text: 'stuff'
end
```

**Comment by jmartin-sul** [comment]
"This code is slightly opaque - it depends on how the foxml gets indexed and then how the Solr index is translated to a display string."

yeah, it's pretty roundabout, but this is how the vast majority of the search and display works in argo (and as i understand it, presenting a UI over solr results is what blacklight was designed to do).  so we index the foxml from fedora to solr (by calling `to_solr` on the fedora object and getting a hash that gets used as a solr doc for indexing).  then search and display from that solr info.  i believe user searches always go to solr, and while there are are a couple things which display directly from fedora, most display also comes from solr (i'm blanking on what comes directly from fedora at the moment).

it would actually be good to document this basic purpose/architecture info somewhere (maybe in the README, maybe a separate file or a github wiki).  it's obvious if you've worked with blacklight before.  but it took me forever to realize what role blacklight played and how everything fit together, because i hadn't worked with blacklight (or rails) at all previously.

"It's not clear to me yet whether there are already tests on the index mapping (if not, need to start there)."

there shouldn't be tests for the index mapping in argo, because that's the responsibility of dor-services (since it implements `to_solr`).  there are lots of tests in dor-services for these mappings, though they certainly aren't comprehensive (and i should turn some post-its on my desk into github tickets for new dor-services tests).  it does feel wrong that argo implicitly has to understand this mapping, even though it doesn't have tests for it.  on the other hand, since argo doesn't do the mapping, there's not really any machinery to be tested to that effect.  i think the right thing to do would be to have integration tests which check to see whether a piece of foxml gets turned into a Dor object with the expected fields, and whether the Dor object can then be solrized and retrieved as expected.  or something.  i'm not sure whether argo is the right home for that or not, but i'm sure that work is outside the scope of this particular bug (though it'd be worth filing an enhancement to come back to and discuss further, because it would be nice to have more insight into how the parts of this mapping can drift, which has obviously happened here).

for now, i'd go with something like jack's suggested test, since that specifically handles the presentation of that solr field, which is what this bug is specifically about.  one of the reasons i hadn't yet added similar tests was that i didn't know how offhand, and i hadn't gotten around to looking into it, since there was such a backlog of things that were in more dire shape.

**Comment by jmartin-sul** [comment]
also, definite :+1: for the fix itself, and thanks for also cleaning up that .html.erb code (which is often even more of a mess than the ruby code, though i guess that's not surprising).

but yeah, simple tests for the UI part would be nice even if the multi-gem-spanning mapping semantics warrant further testing discussion later.

File: spec/features/item_view_metadata_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  scenario 'MD Source is "Symphony"' do
+    visit catalog_path 'druid:kv840rx2720'
+    within '#document-identification-section' do
+      expect(page).to have_css 'dt.blacklight-metadata_source_ssi', text: 'MD Source'
+      expect(page).to have_css 'dd.blacklight-metadata_source_ssi', text: 'Symphony'
+    end
+  end
+
```
Comment (darrenleeweber) [inline spec/features/item_view_metadata_spec.rb]: Sorry but I like the scenario blocks delimited with white space.

**Pull request merged**